onnxruntime/core/mlas/lib/quantize.cpp

```diff
@@ -210,11 +210,217 @@ Return Value:
+            IntegerVectorC = MlasQuantizeLinearPackBytes<DataType>(IntegerVectorC);
+
+#if defined(MLAS_NEON64_INTRINSICS)
+            vst1q_lane_s32((int32_t*)OutputC, IntegerVectorC, 0);
+#else
+            *((int32_t*)OutputC) = _mm_cvtsi128_si32(IntegerVectorC);
+#endif
+
+            InputA += 4;
+            OutputC += 4;
+            N -= 4;
+        }
+    }
+    else {
+        while (N >= 4) {
+            auto IntegerVectorA = MlasQuantizeLinearUnpackBytes<DataType>(MlasBroadcastInt32x4(*((const int32_t*)InputA)));
+            auto IntegerVectorB = MlasQuantizeLinearUnpackBytes<DataType>(MlasBroadcastInt32x4(*((const int32_t*)InputB)));
+            auto FloatVectorC = MlasAddFloat32x4(
+                    MlasDequantizeLinearVector(IntegerVectorA, ScaleVectorA, ZeroPointVectorA),
+                    MlasDequantizeLinearVector(IntegerVectorB, ScaleVectorB, ZeroPointVectorB));
+            auto IntegerVectorC = MlasQuantizeLinearVector(FloatVectorC, ScaleVectorC,
+                    MinimumValueVectorC, MaximumValueVectorC, ZeroPointVectorC);
+            IntegerVectorC = MlasQuantizeLinearPackBytes<DataType>(IntegerVectorC);
+
+#if defined(MLAS_NEON64_INTRINSICS)
+            vst1q_lane_s32((int32_t*)OutputC, IntegerVectorC, 0);
+#else
+            *((int32_t*)OutputC) = _mm_cvtsi128_si32(IntegerVectorC);
+#endif
+
+            InputA += 4;
+            InputB += 4;
+            OutputC += 4;
+            N -= 4;
+        }
+
+    }
+
+    if (N > 0) {
+        auto IntegerVectorA = (LengthA == 1) ?
+                MlasBroadcastInt32x4((int32_t)*InputA) :
+                MlasQuantizeLinearUnpackBytes<DataType>(MlasBroadcastInt32x4(*((const int32_t*)InputA)));
+        auto IntegerVectorB = (LengthB == 1) ?
+                MlasBroadcastInt32x4((int32_t)*InputB) :
+                MlasQuantizeLinearUnpackBytes<DataType>(MlasBroadcastInt32x4(*((const int32_t*)InputB)));
+        auto FloatVectorC = MlasAddFloat32x4(
+                MlasDequantizeLinearVector(IntegerVectorA, ScaleVectorA, ZeroPointVectorA),
+                MlasDequantizeLinearVector(IntegerVectorB, ScaleVectorB, ZeroPointVectorB));
+        auto IntegerVectorC = MlasQuantizeLinearVector(FloatVectorC, ScaleVectorC,
+                MinimumValueVectorC, MaximumValueVectorC, ZeroPointVectorC);
+        IntegerVectorC = MlasQuantizeLinearPackBytes<DataType>(IntegerVectorC);
+
+        uint32_t PackedValueC = 0;
+#if defined(MLAS_NEON64_INTRINSICS)
+        vst1q_lane_s32((int32_t*)&PackedValueC, IntegerVectorC, 0);
+#else
+        *((int32_t*)&PackedValueC) = _mm_cvtsi128_si32(IntegerVectorC);
+#endif
+
+        for (size_t n = 0; n < N; ++n) {
+            *((uint8_t*)OutputC + n) = (uint8_t)PackedValueC;
+            PackedValueC >>= 8;
+        }
+    }
+}
+
 #else
 
 //
 // QuantizeLinear implementation using the C++ runtime.
 //
+template<typename OutputType, int32_t MinimumValue, int32_t MaximumValue>
```

Comment:
merge problem? This template isn't referenced. The MlasQuantizeLinear below is used now instead.

Reply:
Removed. I should noticed this more carefully.